### PR TITLE
[SPARK-5441][pyspark] Make SerDeUtil PairRDD to Python conversions more robust

### PR DESCRIPTION
**Pull request by ghost**
SerDeUtil.pairRDDToPython and SerDeUtil.pythonToPairRDD now both support empty RDDs by checking the result of take(1) instead of calling first which throws an exception.

### CODE REVIEW

**Comment by pwendell** [comment]
Hey thanks for this - mind adding a regression test that fails on the old code?

**Comment by mnazario** [comment]
I've added two regression tests which I made sure failed beforehand and succeed now.

**Comment by JoshRosen** [comment]
Jenkins, this is ok to test.

**Comment by JoshRosen** [comment]
Thanks for adding tests.  This looks good to me, so I'm going to merge it into `master` (1.3.0) and mark it for later backport into `branch-1.2` (I'd commit it now, but we're in the middle of the 1.2.1 voting period right now, so we've placed a hold on merging in that branch until the vote passes).

File: core/src/test/scala/org/apache/spark/api/python/SerDeUtilSuite.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.api.python
+
+import java.io.{ByteArrayOutputStream, DataOutputStream}
+
+import org.apache.spark.{SharedSparkContext, SparkContext}
```
Comment (JoshRosen) [inline core/src/test/scala/org/apache/spark/api/python/SerDeUtilSuite.scala]: Minor nit: we usually place the Spark imports in their own section, separate from third-party library imports like Scalatest.  I'll just fix this myself on merge, but I thought I'd mention it for future patches.

File: core/src/test/scala/org/apache/spark/api/python/SerDeUtilSuite.scala [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.api.python
+
+import java.io.{ByteArrayOutputStream, DataOutputStream}
```
Comment (JoshRosen) [inline core/src/test/scala/org/apache/spark/api/python/SerDeUtilSuite.scala]: Actually, it turns out that these imports are unused.  I'll clean this up, too.

**Comment by JoshRosen** [comment]
I've cherry-picked this to `branch-1.2` (1.2.2).

**Pull request closed without merging**